### PR TITLE
python3Packages.pytest-tap: init at 3.3

### PR DESCRIPTION
**Pull request by Cynerd**
###### Description of changes

https://pypi.org/project/pytest-tap

The handy extension if pytest driven tests are included alongside other test frameworks and TAP is used as a standard output for all of them.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pytest-tap/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  pname = "pytest-tap";
+  version = "3.3";
+  disabled = pythonOlder "3.6";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "5f0919a147cf0396b2f10d64d365a0bf8062e06543e93c675c9d37f5605e983c";
+  };
+
+  propagatedBuildInputs = [ pytest tappy ];
```
Comment (fabaff) [inline pkgs/development/python-modules/pytest-tap/default.nix:19]: `pytest` should not be propagated for `pytest-*` module as it prevents overriding. Move it to `buildInputs`.

File: pkgs/development/python-modules/pytest-tap/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
```
Comment (fabaff) [inline pkgs/development/python-modules/pytest-tap/default.nix:20]: The upstream repo contains [`tests`](https://github.com/python-tap/pytest-tap/tree/main/tests). Whenever possible we should run those tests.

Also, please add `pythonImportsCheck`.

File: pkgs/development/python-modules/pytest-tap/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  meta = with lib; {
+    description = "Test Anything Protocol (TAP) reporting plugin for pytest";
+    homepage = "https://github.com/python-tap/pytest-tap";
```
Comment (fabaff) [inline pkgs/development/python-modules/pytest-tap/default.nix:23]:
```suggestion
    homepage = "https://github.com/python-tap/pytest-tap";
    changelog = "https://github.com/python-tap/pytest-tap/blob/v${version}/docs/releases.rst";
```

File: pkgs/development/python-modules/pytest-tap/default.nix (hunk 1 above)
Comment (Cynerd) [reply pkgs/development/python-modules/pytest-tap/default.nix:19]: I see... That makes sense.

File: pkgs/development/python-modules/pytest-tap/default.nix (hunk 3 above)
Comment (Cynerd) [reply pkgs/development/python-modules/pytest-tap/default.nix:33]: Added

File: pkgs/development/python-modules/pytest-tap/default.nix (hunk 2 above)
Comment (Cynerd) [reply pkgs/development/python-modules/pytest-tap/default.nix:30]: Tests added...

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 207977` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.pytest-tap</li>
    <li>python39Packages.pytest-tap</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/207977#issuecomment-1368210719)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-207977/results/python310Packages.pytest-tap/lib/python3.10/site-packages]$ python
Python 3.10.9 (main, Dec  6 2022, 18:44:57) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pytest_tap
>>> pytest_tap.__version__
'3.3'
```

Looks good to me.

**Pull request merged**